### ISSUE TITLE
Modality fusion method: Cross Attention between features and evidences

### ISSUE THREAD

**Issue by diiinesh** [jonasrohw/mai-project#5]
In this modality fusion method, we try to improve the model by applying cross attention to the features and evidences.
By doing that we hope that we get representations that have encoded more complex releationships between the features and evidences.

The modifications to make in the codebase:

- models.py
-- implementation of the CrossAttention module and StackedCrossAttention module

- utils.py
-- prepare_input(): calling the StackedCrossAttention module and applying cross attention to the features/evidences
-- eval_verite(), train_step(), eval_step(): all of these functions use the prepare_input funtion

- run_experiment.py
-- included new parameters: num_heads_options=4, dropout_options=0.2, num_layers_options=3, 
weight_decay=1e-3
-- increased early_stopping from 10 to 20
-- initialization of the StackedCrossAttention module
-- init weights of the model with xavier weights
-- modified optimizer such that the parameters of the StackedCrossAttention module also gets optimized and 
-- included weight_decay -> mitigates overfitting
-- implemented CyclicLR scheduler that increases the learning rate step by step after each epoch, baseLR = 1e-5 and 
max_lr = 1e-3
